packages/kbn-user-profile-components/src/user_profiles_selectable.tsx

```diff
@@ -90,22 +132,35 @@ export interface UserProfilesSelectableProps
   selectedStatusMessage?(selectedCount: number): ReactNode;
 
   /**
-   * Text for label of clear button.
+   * Label for clear button.
    */
   clearButtonLabel?: ReactNode;
+
+  /**
+   * Label for default options separator.
+   */
+  separatorLabel?: string;
 }
 
+/**
+ * Props of {@link UserProfilesSelectable} component
+ */
+export type UserProfilesSelectableProps = CommonProps &
+  ExclusiveUnion<PropsWithoutNull, PropsWithNull>;
+
 /**
  * Renders a selectable component given a list of user profiles
  */
 export const UserProfilesSelectable: FunctionComponent<UserProfilesSelectableProps> = ({
   selectedOptions,
   defaultOptions,
   options,
+  allowNull = false,
+  showSeparator = !allowNull,
   onChange,
   onSearchChange,
   isLoading = false,
-  singleSelection = false,
+  singleSelection = allowNull,
```

Comment:
Does this mean that if we want to enable the `No assignees` functionality we can only select one user per time?

Reply:
It's just the default so you can select multiple users as well if you want. 

Having said that I don't think multi-selection makes sense when enabling the "unassigned" option. 

A select menu should either be single-select or multi-select. Having both `allowNull` and `singleSelection` enabled is problematic from a usability perspective since selecting "unassigned" will clear selection while selecting any other option will not. 

It also doesn't make a lot of sense when thinking about what the user is trying to achieve. Is it a common user goal to find all cases that both Foo and Bar are assigned to but not cases that only Foo is assigned to? Do we expect users to remember the entire list of assignees about a case and filter on it? I don't think this is a common enough use case that justifies breaking established conventions. It's not even clear whether multi-select narrows down or broadens the selection. (OR vs. AND)

If filtering using multiple assignees is desired I would suggest coming up with a different UI than a simple select menu.

Reply:
@thomheymann: Per our previous Slack conversations on the issue, I completely understand the concern about supporting both multi-assignee selection **and** a `No assignees` option that would deselect any currently selected assignees. It would represent a behavioral inconsistency in the filter assignee interface. This direction was originally suggested because the assumption was that filtering by multiple assignees would use an `and` condition, rather than an `or` condition. Filtering by a case with a selected assignee **AND** `No assignees` would never return any cases. 

However, in a conversation @arisonl and I had with the cases team today, we ultimately decided that we should instead use the `or` condition for the assignee filter, to be consistent with how other filter controls are handled across Kibana. Given that change in filter conditions (and that fact that @arisonl has affirmed the importance of including multi-assignee filtering), I think we can simply show both the `No assignees` option alongside all available users in the assignees filter and allow one or more to be selected (even if one of those selections is `No assignees`). With that condition change, filtering by a case with a selected assignee **OR** `No assignees` would now have the potential to return cases.

If there are concerns that the use of the `or` condition is non-obvious, we can also consider the possibility of also representing that condition in the search input when these filter buttons are interacted with ([examples shown in the EUI docs](https://elastic.github.io/eui/#/forms/search-bar)).

![image](https://user-images.githubusercontent.com/3884767/188718412-4ecf043d-cb65-4eeb-87dd-83244a2ef924.png)

So in summary, my suggestion is to do as follows:

- Assignee filters should support multi-user selection via an `or` filter condition.
- Selecting `No assignees` from the assignee filter options no longer needs to deselect any currently selected assignees, as the condition change to `or` no longer prevents any cases from being returned when selected in conjunction with a user (thus avoiding the behavioral inconsistency mentioned above).

CCing @arisonl, in case he has any additional thoughts.

Reply:
Yeh, I think that makes more sense.

I have changed the behaviour according to new requirements.